### PR TITLE
k8scloudconfig: introduce changelog

### PR DESCRIPTION
**Pull request by kopiczko**
NewTemplate was not used in any of the operators. CloudConfig structs
are created from version package and we should keep changelog to not
become lost.

I ping everybody here as this is quite important to realize.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+### Added
+- Add encruyption config template for API etcd data encryption exeprimental
+  feature.
+- Add tests for template evaluations.
+- Allow adding extra manifests.
+- Allow extract Hyperkube options.
+- Allow setting custom K8s API address for master nodes.
+- Allow setting etcd port.
+- Allow setting etcd port.
```
Comment (puja108) [inline CHANGELOG.md]: double mention

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+and this project's packages adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
+The latest version is considered WIP and it is a subject of change. All other
+versions are frozen. To freeze current version all files are copied to a new
+version directory, and then changes are introduced.
+
+## [v3.0.0] WIP
+
+### Added
+- Add encruyption config template for API etcd data encryption exeprimental
```
Comment (puja108) [inline CHANGELOG.md]: typo: `encryption`

Comment (pipo02mix) [reply CHANGELOG.md]: typo: exeprimental

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+### Changed
+- Unify CloudConfig struct construction.
+- Update calico to 3.0.1.
+- Update hyperkube to v1.9.0.
+- Use vanilla (previously coreos) hyperkube image.
+
+## [v2.0.0]
+
+### Added
+- Disable API etcd data encryption exeprimental feature.
```
Comment (pipo02mix) [inline CHANGELOG.md]: typo: exeprimental

**Comment by kopiczko** [comment]
> how you distinguish between added and changed?

http://keepachangelog.com/en/1.0.0/ - it is also linked in the CHANGELOG.md

> how this file should be updated and when?

I think in every PR that does changes. Excluding maybe things like refactoring.

**Pull request merged**